cloud-custodian/cloud-custodian
# releng/rebase deps 2021 09 22

- releng - use poetry instead of tox in ci
- pytest via module
- requests dropped chardet
- use poetry run to get sphinx in path
- aws - ebs - force result pagination (#6875)
- go minimal
- remove makefile ism
- azure - Change ACI image name (#6888)
- aws - elasticache replication group - tagging support (#6858)
- azure -  session tests fix (#6892)
- use poetry upstream recommended installer, try a shared venv across projects
- fix bash condition
- dir check
- add poetry to path
- core - reduce filter - fix value_regex validation (#6899)
- docs - fix broken external link to point to mailer README instead (#6897)
- disable windows for the momen
- re-enable windows, echo os, setup path, update setuptools/wheel
- export virtualenv env
- use ostype for os dispatch in bash
- separate win32 package set, disable windows temp
- flatten bash in make
- disable mailer
- disable terraform pkg
- add setuptools to c7n_terraform build sys and re-enable
- renable mailer
- preamble the install
- preamble the install x2
- disable log exporter
- update poetry to 1.1.9
- turn on poetry debug output
- remove poetry debug, remove action debug echo, disable failing azure tests
- re-enable log-exporter
- renable windows
- remove timeout as specific to linux/gnu-coreutils .. remove current path ref in test-poetry
- duplicate test-poetry line into action yaml
- path update before cache check
- try source command and relative path
- doh forgot to bash for windows
- simplify test run
- try to figure out cache ids in matrix
- windoze path fun
- restore supported py versions on linux
- on windows skip xdist usage :shrug: re-enable linux py version matrix
- restore azure keyvault tests via add of missing env var, sanity check venv contents, explict disable virtualenv creation
- delay poetry config till its in path
- doh on windows its scripts not bin
- minimize to isolate windows issues
- use bash
- windows can't replace binaries being executed
- we really need to upgrade pip, worth considering ditching windows in ci?
- pip upgrade that worksarounds windows
- typo fix
- rename ci actions/workflow and fix odd char on pip up
- so many typos
- backslash == escape .. why windows.. why!
- transient disable
- transient more
- use separate windows dep spec
- omit separate windows install bit
- add back tool/provider tests
- skip policystream tests on windows
- restore static analyzer, code cov, and add makefile test codecov target
- sanity check cov env vars
- sanity check cov env vars
- omit commented lines for yaml out of codecov debug
- ditch make for run test step
- export debug value for codecov env var
- codecov packages enum
- fix typo
- ident cov runs via echo
- run coverage gen xml
- restore matrix os and versions
- nit - matrix ordering
- restore doc build in ci
- use poetry run for doc build, minimize build matrix
- doc build - unix paths, use custodian version for cache health check
- restore full matrix
- try without poetry run to see if we can better streaming results
- sphinx build cache, restore cross-provider doc example checking
- remove makefile target for poetry windows
- fix makefile test poetry targets
- auto diff first changed resource file
- flake8
- docs - cache generated files
- docs - turn on sphinx build warn as err, fix extant warn, update pygments, terraform as literal
- trim tox config
- dep updates
- gcp dep update fixes
